### PR TITLE
Integration performances

### PR DESCRIPTION
**Pull request by hasnatali1947**
In this PR, I have done the following:
N+1 problem is solved when fetching all posts and their comments for a user.

- Before:
image
image

- After
image

Use Capybara to write integration tests for each view includes:

User index page:

user show page:

User post index page:

Post show page:

### CODE REVIEW

File: spec/intergration/post_index_spec.rb
Comment (fobadara) [inline spec/intergration/post_index_spec.rb:1]: [REQUIRED] You are missing tests that assert you can see a section for pagination if there are more posts than fit on the view.

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1,25 +1,37 @@
...
 </div>
+
+<!-- Add the placeholder here -->
+<ul class="photo">
+  <!-- Placeholder content goes here -->
+</ul>
+
 <ul class="posts">
-<% @user.posts.each do |post| %>
+  <% @posts.includes(:author, :comments).each do |post| %>
```
Comment (fobadara) [inline app/views/posts/index.html.erb:20]: [OPTIONAL] I can see you using posts here. But I can't see any @posts variable in the index of your users controller

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @hasnatali1947, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Use of description blocks
✅ Good first attempt


## Required Changes ♻️
_Check the comments under the review._
I had to include the comment here because it has to do with multiple files

[REQUIRED] I noticed some error when I run your test
image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by hasnatali1947** [comment]
image

all test are passed 👍

**Review by algerina — APPROVED** [review]
Hi @hasnatali1947  ,

Great job so far!  

May I highlight the following: 
- [x] Your implemented all the requirements of the previous TSE✔️ 
- [x] Used Gitflow
- [x]  Work documented professionally



## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**